### ISSUE TITLE
don't require inheriting from mbgl.FileSource

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/node-mapbox-gl-native#141]
Upstream followup to https://github.com/mapbox/tilelive-gl/issues/30. The current API is:

```
var fileSource = new mbgl.FileSource();
fileSource.request = function(req) {
  // TODO
};
fileSource.cancel = function(req) {
  // TODO
};
var map = new mbgl.Map(fileSource);
```

Instead, this should be possible:

```
var map = new mbgl.Map({
  request: function(req) {
    // TODO
  },
  cancel: function(req) {
    // TODO
  }
});
```

And eventually we want to move to an API like:

```
var map = new mbgl.Map({
  readStyle: function(url, callback) {
    // TODO
  },
  readSource: function(url, callback) {
    // TODO
  }
  ...
});
```

Gist of the implementation here is that `NodeFileSource` should no longer inherit from `node::ObjectWrap`, instead it gets a `Handle<v8::Object>` (pointing to the object passed by the user) in the constructor, and dispatches request/cancel to this handle rather than `NanObjectWrapHandle(this)`.

**Comment by tmpsantos** [comment]
:+1:

**Comment by mikemorris** [comment]
I think I almost have this working in the [`internal-filesource`](https://github.com/mapbox/node-mapbox-gl-native/compare/internal-filesource) branch, but I can't figure out the right target to pass to `NanMakeCallback` in `NodeFileSource::processAdd` and `NodeFileSource::processCancel`...

**Comment by mikemorris** [comment]
Fixed in #143
